test/integration/buckets-client-validation.test.js

```diff
@@ -147,6 +186,62 @@ test('buckets client validation', testOpts, function (suite) {
         });
     });
 
+    VALID_LIST_BUCKETS_QUERY_OPTS.forEach(function (query_opts) {
+        test(f('validating list buckets: %j', query_opts), function (t) {
+            var s = client.createListBucketsStream({query: query_opts});
+
+            s.on('readable', function onReadable() {
+                while (s.read() !== null) {
+                    // exhaust the stream
+                }
+            });
+
+            s.once('error', function onError(err) {
+                t.ifError(err);
+                t.end();
+            });
+
+            s.once('end', function onEnd() {
+                t.ok('end seen');
+                t.end();
+            });
+        });
+    });
+
+    INVALID_LIST_BUCKETS_QUERY_OPTS.forEach(function (query_opts) {
+        test(f('validating invalid list buckets: %j', query_opts),
+            function (t) {
+
+            var s;
+
+            try {
+                s = client.createListBucketsStream({query: query_opts});
+            } catch (e) {
+                t.ok(e, f('error in client validation: %s', e.message));
+                t.end();
+                return;
+            }
+
+            s.on('readable', function onReadable() {
+                while (s.read() !== null) {
+                    // exhaust the stream
+                }
+            });
+
+            s.once('error', function onError(err) {
+                assert.ok(err, 'error seen: ' +
```

Comment:
This should be `t.ok` rather than `assert.ok`, right? Currently these are showing this TAP output:

```
    # Subtest: validating invalid list buckets: {"delimiter":-1}
        1..0
    ok 29 - validating invalid list buckets: {"delimiter":-1} # time=49.494ms

    # Subtest: validating invalid list buckets: {"delimiter":false}
        1..0
    ok 30 - validating invalid list buckets: {"delimiter":false} # time=47.948ms

    # Subtest: validating invalid list buckets: {"delimiter":"AB"}
        1..0
    ok 31 - validating invalid list buckets: {"delimiter":"AB"} # time=49.037ms
```

Reply:
yep! will change